src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -4,6 +4,9 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+#if NET6_0_OR_GREATER
+using System.Linq;
```

Comment:
I just realized that STJ can't take a dependency on System.Linq. I would recommend then to just add a simple check for `ICollection<T>` for all target frameworks.